docs/architecture.md

```diff
@@ -121,33 +119,58 @@ FxDesktopRelier is used when interfacing with Firefox Desktop Sync.
 
 #### Authentication Brokers
 
-##### Base
+##### base
 BaseAuthenticationBroker is the generic interface for other brokers.
 
-##### OAuth
-OAuthAuthenticationBroker provides functionality common to all OAuth related flows. The OAuthAuthenticationBroker saves OAuth data before verification, generates OAuth results, and sends the OAuth result to the OAuth relying party.
+##### fx-desktop-v1
+Interfaces with Sync on Fx Desktop. A custom protocol allows the content server to query the browser for information and send results back.
 
-###### Redirect
-RedirectAuthenticationBroker is used for redirect based OAuth flows. The RedirectAuthenticationBroker sends results to the OAuth relying party via the browser's URL.
+##### fx-desktop-v2
+v2 of the Fx Desktop authentication broker, it speaks to the browser using WebChannels.
 
-###### IFrame
-IFrameAuthenticationBroker is used for iframe based OAuth flows. This broker communicates with the OAuth relying party via postMessage. The IFrame broker ensures the parent IFRAME is the same domain as the domain registered for the `client_id` passed on startup.
+##### fx-desktop-v3
+v3 of the Fx Desktop authentication broker, it allows the uid of a user to change if the user's account has been deleted.
 
-###### WebChannel
-WebChannelAuthenticationBroker is used for communicating with browser based services like Loop. WebChannelBroker communicates with the browser via postMessage and special events.
+##### fx-fennec-v1
+Interfaces with Sync on Fennec. Uses WebChannels to communicate with the browser.
 
-##### FxDesktop
-FxDesktopAuthenticationBroker interfaces with Firefox Sync. A custom protocol allows the content server to query the browser for information and send results back.
+##### fx-firstrun-v1
+Used by FxDesktop to embed FxA within an iframe on the firstrun page. Communicates with the browser using web channels.
+
+##### fx-firstrun-v2
+v2 of the firstrun authentication broker, used to enable "Choose what to Sync".
+
+##### fx-ios-v1
+Interfaces with Sync on Fx for iOS. Uses the same custom protocol as FxDesktop v1.
+
+##### fx-sync
+Generic Sync broker. Used to define behaviors for other Sync brokers, and if
+the user verifies a Sync sign up in a 2nd browser.
+
+##### fx-sync-channel
+Base for brokers that communicate with Firefox.
+
+##### fx-sync-web-channel
+Generic interface used for Syncing with browsers that support web channels.
```

Comment:
> Sure, then are these 3 just for Firefox desktop then?

fx-sync is used for all Sync related interactions, signup and verification, no matter which browser.

fx-sync-web-channel is used by fx-desktop-v2+, fx-firstrun-*, fx-fennec-v1

fx-sync-channel is used for any integration that expects to actually communicate with the browser. This is the one I liked the least, and I was most unsure of. I considered making it a mixin to other modules, but didn't in the end because that would mean modules that mix in those functions wouldn't be able to override the functions in the mixin. I thought about not adding a new module and using the old fx-sync for everything, and passing in configuration that says "don't actually send any messages" so we could use fx-sync in the verification tab. I tried keeping fx-sync as is and creating a new broker, "fx-sync-verification" which took fx-sync's behaviors and that's it, but that looked really strange to me. This seemed like the least bad approach I came up with, though maybe there is merit in `fx-sync-verification`


The idea is that fx-sync is generic and defines behaviors that are common across all Sync integrations, for signup and verification.

fx-sync-channel is used for integrations